### PR TITLE
 Issue #3232778 - error when upgrading to 10.2.5 on drush updatedb

### PR DESCRIPTION
**Pull request by nijuley**
## Problem
The following error is being thrown while upgrading from 9.18 to 10.2.x or latest version, preventing from commencing the upgrade process.
Error: Call to a member function label() on null in `/var/www/html/web/modules/contrib/group/src/Plugin/GroupContentEnablerBase.php on line 171 /var/www/html/web/modules/contrib/group/src/Entity/Storage/GroupContentTypeStorage.php(115): Drupal\group\Plugin\GroupContentEnablerBase->getContentTypeLabel()`

## Solution

In social_group.install, function social_group_update_8906(), instead of assuming that there are a public_group, open_group and close_group, use group_type storage to load existing group_types, then use group_type_ids to create group managers by default.
use patch file from `https://www.drupal.org/project/social/issues/3232778`

## Issue tracker
https://www.drupal.org/project/social/issues/3232778

## Theme issue tracker
N/A

## How to test

- [ ] Install old version (possibly 10.0.x or 10.1.x)
- [ ] Configure open social to have only 2 groups, delete public_group for instance
- [ ] Export all config and settings related
- [ ] Upgrade to latest version and perform drush updatedb
- [ ] This error happens

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] If applicable (I.E. new hook_updates) the update path from previous versions (major and minor versions) are tested
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Screenshots
N/A

## Release notes

N/A

## Change Record
N/A

## Translations
N/A

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2888-bwd767py8iubjrmfzplyb62kdoypag8l.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/625d97faef17c6f56950b4d3

File: composer.json [hunk 1]
```diff
@@ -244,6 +244,14 @@
         "drupal/social": "self.version"
     },
     "config": {
-        "sort-packages": true
+        "sort-packages": true,
```
Comment (navneet0693) [inline composer.json:247]: This change shouldn't be part of the profile's composer.json. It should be part of "root" composer.json of the created project using "social" profile.

File: composer.lock [hunk 2]
```diff
@@ -0,0 +1,15125 @@
+{
```
Comment (navneet0693) [inline composer.lock:1]: composer.lock file should never be a part of distribution profiles.

File: modules/social_features/social_group/social_group.install [hunk 3]
```diff
@@ -716,19 +716,13 @@ function social_group_update_8905() {
...
-
-  $group_types = [
-    'public_group',
-    'open_group',
-    'closed_group',
-  ];
-
-  foreach ($group_types as $group_type) {
-    $config = $config_factory->getEditable('group.type.' . $group_type);
+  $group_types = \Drupal::entityTypeManager()->getStorage('group_type')->loadMultiple();
```
Comment (navneet0693) [inline modules/social_features/social_group/social_group.install:719]: Just for reviewers understanding:

At the time of Open Social 8.x, we only shipped with public, closed, and open group types. That is why we never considered writing the update hook for "other" group types as well. We only wanted to focus on default groups shipped by Open Social.

**Review by navneet0693 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**